### ISSUE TITLE
Using --export influxdb and can't see docker stats in grafana

### ISSUE THREAD

**Issue by sUyMur** [nicolargo/glances#1839]
**Describe the bug**
When using OPTS "-w" I can see all my container and their stats. When using --export influxdb (-q -t 6) I cannot.

**To Reproduce**
1. Start Glances using -w and see all containers
2. Start Glances without -q and see all containers
3. Start Glances with the followings options '--export influxdb -q -t 6' and I can't see all containers (in grafana) but could 2 days ago.

**Expected behavior**
Some type of information about my container in grafana.

**Desktop (please complete the following information):**
 - OS: DietPI 7.0.2 (Debian) 
 - uname -r: 5.10.17-v7l+
 - Glances Version 
```
Glances v3.1.6.2 with PsUtil v5.8.0
Log file: /home/pi/.local/share/glances/glances.log
```
 - Glances logs file
```
2021-04-07 17:29:05,306 -- INFO -- Start GlancesStandalone mode
2021-04-07 17:29:05,756 -- WARNING -- Wifi lib is not compliant with Python 3, Wifi plugin is disabled
2021-04-07 17:29:05,983 -- INFO -- Stats will be exported to InfluxDB server: http://localhost:8086
2021-04-07 17:29:07,365 -- INFO -- Quiet mode is ON, nothing will be displayed
2021-04-07 19:34:36,449 -- INFO -- Start Glances 3.1.6.2
2021-04-07 19:34:36,450 -- INFO -- CPython 3.7.3 and psutil 5.8.0 detected
2021-04-07 19:34:38,363 -- INFO -- Start Glances 3.1.6.2
2021-04-07 19:34:38,363 -- INFO -- CPython 3.7.3 and psutil 5.8.0 detected
2021-04-07 19:34:41,871 -- INFO -- Start Glances 3.1.6.2
2021-04-07 19:34:41,871 -- INFO -- CPython 3.7.3 and psutil 5.8.0 detected
```

**Additional context**
Installed glances locally and used the following to start it: 
`/usr/bin/python3 /home/pi/.local/bin/glances -C /home/pi/.config/glances.conf --export influxdb -q -t 6 &`
This worked till two days ago.

Since I thought I broke my local install or messed up a script I cloned the git.
I used your docker which didn't work as easily so I just build the docker myself and fired it up.
Modified the Dockerfile in line 25:
from: `RUN pip install glances[all]`
to: `RUN pip install glances[all] && pip install influxdb`
log: 
docker build --tag glances:armv7 .
```
Sending build context to Docker daemon  7.168kB
Step 1/9 : ARG ARCH=
Step 2/9 : FROM ${ARCH}python:3-slim-buster
 ---> 0bbb47dfe6c5
Step 3/9 : ENV DEBIAN_FRONTEND noninteractive
 ---> Using cache
 ---> 3ec37cea46a6
Step 4/9 : RUN   apt-get update           &&   apt-get install -y            curl                gcc                 lm-sensors          wireless-tools      iputils-ping &&   rm -rf /var/lib/apt/lists/*
 ---> Using cache
 ---> b11a4523d517
Step 5/9 : ARG VCS_REF
 ---> Using cache
 ---> 0a656623a9a9
Step 6/9 : RUN pip install glances[all] && pip install influxdb
 ---> Using cache
 ---> 78bf42f3812f
Step 7/9 : WORKDIR /glances
 ---> Using cache
 ---> 28595585fac6
Step 8/9 : EXPOSE 61209 61208
 ---> Using cache
 ---> c7d9af695231
Step 9/9 : CMD python3 -m glances -C /glances/conf/glances.conf $GLANCES_OPT
 ---> Using cache
 ---> 7b35af6c74c2
Successfully built 7b35af6c74c2
Successfully tagged glances:armv7
```

Used docker-compose.yaml:
```
  glances:
    image: glances:armv7 
    container_name: glances
    restart: unless-stopped
    pid: host
    network_mode: bridge 
    ports:
      - 61208:61208 
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
      - /mount/.config/glances/:/glances/conf/
    environment:
      - "GLANCES_OPT=--export influxdb -q -t 6"
#      - "GLANCES_OPT=-w" #To check if it actually works and shows information about container. It does.
```
Important parts of config from /mount/.config/glances/glances.conf:
```
[influxdb]
# Configuration for the --export influxdb option
# https://influxdb.com/
host=$serverip
port=8086
protocol=http
user=
password=
db=glances
# comments
prefix=localhost

[docker]
disable=False
#cpu_careful=50
# Thresholds for CPU and MEM (in %)
#cpu_warning=70
#cpu_critical=90
#mem_careful=20
#mem_warning=50
#mem_critical=70
#
# Per container thresholds
#containername_cpu_careful=10
#containername_cpu_warning=20
#containername_cpu_critical=30
#
# By default, Glances only display running containers
# Set the following key to True to display all containers
all=False
```

I can see information about CPU / HDD / network just fine. But no more container information.

Grafana Select for CPU:
`SELECT mean("min1") FROM "localhost.load" WHERE $timeFilter GROUP BY time($__interval) fill(none)`

Grafana Select for (a) Docker:
`SELECT mean("$container.cpu_percent") FROM "localhost.docker" WHERE $timeFilter GROUP BY time($__interval) fill(none)`
or specific, so it does not depend on vars (also, all of them are provided through grafana as suggestion):
`SELECT mean("NordVPN.cpu_percent") FROM "localhost.docker" WHERE $timeFilter GROUP BY time($__interval) fill(null)`
and just to be sure it's really online: `docker ps | grep NordVPN`  
yes it is `up 4 hours (healthy)`

I'm sorry for the wall of text, let me know if I missed something or you need anything else.
I guess I'm just not using grafana / influxdb right, but I wonder what changed in the last few days.

Edit: I use this grafana dashboard: https://grafana.com/grafana/dashboards/2387

**Comment by sUyMur** [comment]
They are back. 

I didn't change anything. I just have a daily job to stop all containers and restart them after backing up important data.

Issue can be closed I guess?

**Comment by sUyMur** [comment]
And they are gone again. 
After the last startup this morning docker stats seize to exist. All others are fine. 

Guess I will have a look again tomorrow.

**Comment by nicolargo** [comment]
I just make a test from scratch on my system and stats' containers are exported like a charm.

I will let this configuration run for a while.

**Comment by sUyMur** [comment]
I have redone the containers too, but can't figure it out for now.
What i could see with `environment:      - "GLANCES_OPT=-w"` was all the docker stats too, just not in grafana.
I installed chronograf and had a look at the data over there too. Same thing, all the system stats (when using -q -t 6), but no docker stats.
Thinking it could've been permissions I started glances on the host itself - nothing changed.

I then installed telegraf and fed the data to a new DB. For now, since this setup provides me with all the data I need, I will use it for now. 

If at some point I might find the reason for the odd behaviour regarding the export of docker stats to influxdb, I will comment.
